acapy_plugin_toolbox/holder/v0_1/__init__.py

```diff
@@ -109,6 +108,18 @@ async def setup(
     )
 
 
+def send_condition(principal: Principal, record: PresExRecord, approved: set):
+    if "all-credentials" in principal.privileges:
+        return True
+    elif "limited-credentials" in principal.privileges:
+        cred_def_id_restrictions = set(record.credential_definition_id)
+        if cred_def_id_restrictions.issubset(approved):
+            return True
+        else:
+            return False
+    return False
+
+
```

Comment:
Okay, I think we have a bit of confusion here. The send condition of the credential offer received notification will differ from the send condition of the presentation request received notification so we should actually be able to reuse the same `send_condition` for both.

For the credential case, it was a matter of determining whether the cred def ID was in the approved set. But, for the presentation case, we have to go through the same cred def id restriction extraction process we went through for the other presentation filtering pieces and then make sure the set of extracted cred def ids is a subset of the approved set.

Reply:
I was missing the keyword "not" in my previous comment. We cannot reuse the same `send_condition` for both credential and presentation cases.

Reply:
Ah, thanks for this explanation! I think I'm still a bit confused- is this closer?

This `send_condition` function just determines whether the cred def ID is in the approved set, so this is the condition we use for the credential case (in `issue_credential_event_handler()`):
`    def send_condition(principal: Principal):
        # TODO generalize privilege?
        if "limited-credentials" in principal.privileges:
            return record.credential_definition_id in approved
        if "all-credentials" in principal.privileges:
            return True
        return False`

Whereas this `send_condition` function checks that the set of extracted cred def ids is a subset of the approved set, so this is  the condition we use for the presentation case (in `present_proof_event_handler()`)?
`def send_condition(principal: Principal, record: PresExRecord, approved: set):
    if "all-credentials" in principal.privileges:
        return True
    elif "limited-credentials" in principal.privileges:
        cred_def_id_restrictions = set(record.credential_definition_id)
        return cred_def_id_restrictions.issubset(approved)
    else:
        return False`